debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java

```diff
@@ -230,17 +235,41 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
                 producer.send(
                         producer.messageBuilder().addData(getBytes(value)).build(),
                         confirmationStatus -> {
+                            try {
+                                if (confirmationStatus.isConfirmed()) {
+                                    committer.markProcessed(record);
+                                }
+                                else {
+                                    LOGGER.error("Failed to confirm message delivery for event '{}'", record);
+                                    hasError.set(true);
+                                }
+                            }
+                            catch (Exception e) {
+                                LOGGER.error("Failed to process record '{}': {}", record, e.getMessage());
+                                hasError.set(true);
+                            }
+                            finally {
+                                latch.countDown();
+                            }
                         });
             }
             catch (StreamException e) {
                 throw new DebeziumException(e);
             }
+        }
 
-            committer.markProcessed(record);
+        if (!latch.await(producerConfirmTimeout, TimeUnit.SECONDS)) {
```

Comment:
IMHO we need a new config option for this wait set to a larger default value as timeout for sending a single message could be different to sending a batch.

Reply:
done